src/cloud-providers/util.go

```diff
@@ -142,3 +143,39 @@ func WriteUserData(instanceName string, userData string, dataDir string) (string
 	// Return the file path
 	return filePath, nil
 }
+
+// Verify SSH public key file
+// Check the permissions and the content of the file to ensure it is a valid SSH public key
+func VerifySSHKeyFile(sshKeyFile string) error {
```

Comment:
with the change to mkosi images, we won't have the option to provision SSH keys anymore, since this functionality relies on cloud-init and writeable root filesystems

Reply:
True for CoCo and that's why I added it to the ConfigVerifier method which is disabled by default. For non CoCo (regular peer-pods) this at least gives an option to ensure using valid public key. Is that acceptable ?